src/components/environment-picker.tsx

```diff
@@ -30,15 +31,12 @@ export function EnvironmentPicker(props: EnvironmentPickerProps): JSX.Element {
 
   const labelId = `${props.id}-label`;
 
-  return props.environmentList.length === 1 ? (
-    <TextField
-      label={props.label}
-      variant="outlined"
-      value={props.value}
-      name={props.name}
-      InputProps={{ readOnly: true }}
-    />
-  ) : (
+  // If exactly one environment is present, do not display an environment UI element.
```

Comment:
Are we still setting the value when the form is submitted?

Reply:
Yes; the value is still set in the model, even though nothing is visible in the UI. The model is used to compose the API call on form submission.